src/main/scala/codacy/tailor/Tailor.scala

```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
 
   private lazy val configFileNames = Set(".tailor.yml")
 
-  override def apply(source: Source.Directory,
-                     configuration: Option[List[Pattern.Definition]],
-                     files: Option[Set[Source.File]],
-                     options: Map[Options.Key, Options.Value])(
-      implicit specification: Specification): Try[List[Result]] = {
-    Try {
+  private def nativeConfigurationFromSource(source: Source.Directory): Option[List[String]] = {
+    configFileNames
+      .map(name => Try(File(source.path) / name))
+      .collectFirst {
+        case Success(file) if file.isRegularFile =>
+          List("-c", file.toJava.getAbsolutePath)
+      }
+  }
 
-      lazy val nativeConfig = configFileNames
-        .map(name => Try(File(source.path) / name))
-        .collectFirst {
-          case Success(file) if file.isRegularFile =>
-            List("-c", file.toJava.getAbsolutePath)
-        }
+  private def patternsToUseFromCodacyConfig(
+      configuration: Option[List[Pattern.Definition]]
+  )(implicit specification: Specification): Option[List[String]] = {
+    val patternsToLintOpt = configuration.withDefaultParameters
+
+    patternsToLintOpt.map {
+      case patternsToLint if patternsToLint.nonEmpty =>
+        val patternIds = patternsToLint.map(_.patternId)
+        val parameters =
+          patternsToLint.flatMap(_.parameters).flatten.flatMap { parameter =>
+            List(s"--${parameter.name}", paramValueToJsValue(parameter.value).toString)
+          }
+        List("--only=" + patternIds.mkString(",")) ++ parameters
+      case _ => List.empty[String]
+    }
+  }
 
-      val filesToLint: List[String] = files.fold(List(source.path)) { paths =>
-        paths.map(_.toString).toList
-      }
+  private def lintingConfiguration(source: Source.Directory, configuration: Option[List[Pattern.Definition]])(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [lintingConfiguration should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)